### ISSUE TITLE
scatter job instance types can't be overridden at runtime

### ISSUE THREAD

**Issue by mlin** [dnanexus/dxWDL#96]
@orodeh we discussed this verbally, and it's NOT a high priority IMO, but just to document it here:

We might `dx run a-dxWDL-workflow --instance-type xxxx` and expect it to override the instance type for all jobs in the workflow. For example, we are running unit tests on the workflow and want it to use small machines to reduce costs and scheduling delays just for that purpose.

This works for linear workflows, but not for scatter jobs. The subjobs launched by the scatter controller applet end up running on the compile-time-specified instance type.

**Comment by nishethnexus** [comment]
@mlin  Can this be problem be solved by making instance-type in scatter step as an input with default value ? So user can override it at runtime if needed.

**Comment by mlin** [comment]
@nishethnexus Yes there are multiple workarounds for this, it is not really a priority to address as I mentioned. But it generally is the case if you `dx run some-workflow --instance-type xxxx ...` for a (non-WDL) workflow, it overrides the instance type for all entry points and subjobs, so the current behavior is a bit surprising. That override is useful so that various jobs in a test quickly reuse the same instance type rather than waiting for availability of several different ones.
